### DISCUSSION TITLE
calling AWS Amplify using IAM credentials

### DISCUSSION THREAD

**Discussion by menkari** [graphql-dotnet/graphql-client#492] [Q&A]
I've been looking through examples online to see how best to sign a request using AWS Sig V4 (Access Key and Secret) in order to call a GraphQL endpoint in a server to server communication.

The example I've used passes through a derived class as the HttpMessageHandler which seems to work correctly, however, the graphQL response returns the following error message:

`The request signature we calculated does not match the signature you provided. Check your AWS Secret Access Key and signing method. Consult the service documentation for details.`

I know the access key and secret work as I've used the awssdk JS libraries to sign similar requests in node. I'm tinkering with it today, but wondering if anyone else has had this issue and solved it sufficiently for use with the `HttpMessageHandler`?. I've looked at packages like this one:

https://www.nuget.org/packages/AwsSignatureVersion4/

and 

https://github.com/graphql-dotnet/graphql-client/issues/121

but it looks to be an older answer and doesn't work out of the box - I'm tinkering with this as I write this and hopefully get somewhere.

Any help appreciated.

**Comment by sungam3r** [comment]
@menkari I trensferred this issue from the core repo here.
